kbfsgit/runner_test.go

```diff
@@ -1004,3 +1016,57 @@ func TestRunnerWithKBFSReset(t *testing.T) {
 		rootNode.GetFolderBranch().Tlf, nil, keybase1.MDPriorityGit)
 	require.NoError(t, err)
 }
+
+func testHandlePushBatch(t *testing.T, ctx context.Context,
+	config libkbfs.Config, git, refspec, tlfName string) commitsByRefName {
+	var input bytes.Buffer
+	var output bytes.Buffer
+	r, err := newRunner(ctx, config, "origin",
+		fmt.Sprintf("keybase://private/%s/test", tlfName),
+		filepath.Join(git, ".git"), &input, &output, testErrput{t})
+	require.NoError(t, err)
+
+	args := [][]string{{refspec}}
+	commits, err := r.handlePushBatch(ctx, args)
+	require.NoError(t, err)
+	return commits
+}
+
+func TestRunnerHandlePushBatch(t *testing.T) {
+	ctx, config, tempdir := initConfigForRunner(t)
+	defer libkbfs.CheckConfigAndShutdown(ctx, t, config)
+	defer os.RemoveAll(tempdir)
+
+	git, err := ioutil.TempDir(os.TempDir(), "kbfsgittest")
+	require.NoError(t, err)
+	defer os.RemoveAll(git)
+
+	// Setup the repo
+	h, err := libkbfs.ParseTlfHandle(
+		ctx, config.KBPKI(), config.MDOps(), "user1", tlf.Private)
+	require.NoError(t, err)
+	_, err = libgit.CreateRepoAndID(ctx, config, h, "test")
+	require.NoError(t, err)
+
+	t.Log("Make a new local repo with one commit, and push it. " +
+		"We expect this to return no commits, since it should push the " +
+		"whole repository.")
+	makeLocalRepoWithOneFileCustomCommitMsg(t, git, "foo", "hello", "", "one")
+	commits := testHandlePushBatch(t, ctx, config, git,
+		"refs/heads/master:refs/heads/master", "user1")
+	require.Empty(t, commits)
+
+	addOneFileToRepoCustomCommitMsg(t, git, "foo2", "hello2", "two")
+	commits = testHandlePushBatch(t, ctx, config, git,
+		"refs/heads/master:refs/heads/master", "user1")
+	require.Len(t, commits, 1)
+	require.Len(t, commits["refs/heads/master"], 1)
```

Comment:
It would be great if this test checked the commit messages too, otherwise there's no reason for those `CustomCommitMsg` methods.